### PR TITLE
[chore] add generated tests for receivers

### PR DESCRIPTION
**Pull request by atoulme**
Continuation of the work on https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/27849 - now opening a way to create lifecycle tests for receivers.

### CODE REVIEW

File: cmd/mdatagen/templates/component_test.go.tmpl [hunk 1]
```diff
@@ -224,3 +230,72 @@ func Test_ComponentLifecycle(t *testing.T) {
...
+			firstRcvr, err := test.createFn(context.Background(), receivertest.NewNopCreateSettings(), cfg)
+			require.NoError(t, err)
+			host := newAssertNoErrorHost(t)
+			require.NoError(t, err)
+            require.NoError(t, firstRcvr.Start(context.Background(), host))
+            require.NoError(t, firstRcvr.Shutdown(context.Background()))
+			secondRcvr, err := test.createFn(context.Background(), receivertest.NewNopCreateSettings(), cfg)
+            require.NoError(t, err)
+            require.NoError(t, secondRcvr.Start(context.Background(), host))
+            require.NoError(t, secondRcvr.Shutdown(context.Background()))
```
Comment (dmitryax) [inline cmd/mdatagen/templates/component_test.go.tmpl:297]:
```suggestion
			require.NoError(t, firstRcvr.Start(context.Background(), host))
			require.NoError(t, firstRcvr.Shutdown(context.Background()))
			secondRcvr, err := test.createFn(context.Background(), receivertest.NewNopCreateSettings(), cfg)
			require.NoError(t, err)
			require.NoError(t, secondRcvr.Start(context.Background(), host))
			require.NoError(t, secondRcvr.Shutdown(context.Background()))
```

File: cmd/mdatagen/templates/component_test.go.tmpl [hunk 2]
```diff
@@ -224,3 +230,72 @@ func Test_ComponentLifecycle(t *testing.T) {
 	}
 }
 {{ end }}
+
+{{ if isReceiver }}
+func Test_ComponentLifecycle(t *testing.T) {
```
Comment (dmitryax) [inline cmd/mdatagen/templates/component_test.go.tmpl:235]: Doesn't seem like it's generated for the splunkhecreceiver?

File: receiver/splunkhecreceiver/receiver_test.go [hunk 3]
```diff
@@ -456,7 +456,7 @@ func Test_splunkhecReceiver_TLS(t *testing.T) {
 		require.NoError(t, r.Shutdown(context.Background()))
 	}()
 
-	mh := newAssertNoErrorHost(t)
+	mh := newAssertNoErrHost(t)
```
Comment (dmitryax) [inline receiver/splunkhecreceiver/receiver_test.go:459]: Unrelated changes?

File: cmd/mdatagen/templates/component_test.go.tmpl (hunk 2 above)
Comment (dmitryax) [reply cmd/mdatagen/templates/component_test.go.tmpl:235]: Don't we have a CI check to validate whether it's properly generated? If so we need to add another `git diff` check in `test-and-build.yml::checks` job.

File: receiver/splunkhecreceiver/receiver_test.go (hunk 3 above)
Comment (atoulme) [reply receiver/splunkhecreceiver/receiver_test.go:459]: this is related, there is a name collision on `newAssertNoErrorHost` otherwise for this component.

File: cmd/mdatagen/templates/component_test.go.tmpl (hunk 2 above)
Comment (atoulme) [reply cmd/mdatagen/templates/component_test.go.tmpl:235]: It doesn't because the file is not even checked in (adding it now) and therefore ignored by git diff.

**Review by dmitryax — APPROVED** [review]
